test/components/views/context_menus/MessageContextMenu-test.tsx

```diff
@@ -13,35 +13,220 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
+/* eslint-disable @typescript-eslint/no-var-requires */
 
 import React from 'react';
-import { mount } from 'enzyme';
+import { mount, ReactWrapper } from 'enzyme';
 import { EventStatus, MatrixEvent } from 'matrix-js-sdk/src/models/event';
 import { Room } from 'matrix-js-sdk/src/models/room';
 import { PendingEventOrdering } from 'matrix-js-sdk/src/matrix';
 import { ExtensibleEvent, MessageEvent, M_POLL_KIND_DISCLOSED, PollStartEvent } from 'matrix-events-sdk';
+import { Thread } from "matrix-js-sdk/src/models/thread";
 
 import * as TestUtils from '../../../test-utils';
-import MessageContextMenu from '../../../../src/components/views/context_menus/MessageContextMenu';
 import { MatrixClientPeg } from '../../../../src/MatrixClientPeg';
+import RoomContext, { TimelineRenderingType } from "../../../../src/contexts/RoomContext";
+
+const PATH_TO_STRING_UTILS = "../../../../src/utils/strings";
+const PATH_TO_EVENT_UTILS = "../../../../src/utils/EventUtils";
+
+jest.mock(PATH_TO_STRING_UTILS);
+jest.mock(PATH_TO_EVENT_UTILS);
+
+const { copyPlaintext, getSelectedText } = require(PATH_TO_STRING_UTILS);
```

Comment:
Why do you `require` these?
```
import { copyPlaintext, getSelectedText } from "../../../../src/utils/strings";
jest.mock("../../../../src/utils/strings");
```
Will work the same without losing the types to `require`.
To apply the mock returns without tsc complaining you can wrap the function name with `mocked` from js-mock, which adds a MockedObject wrapper around the function
```
mocked(getSelectedText).mockReturnValue(text);
```

Reply:
do you want some of the utils to be unmocked in some tests, mocked in others? Or you just want different mock behaviour per test?

Reply:
I want them to be mocked by test

Reply:
So if I understand correctly you want different mock return values per test?
That's completely possible with the normal import and mock. To keep behaviour independent per test case you can reset all the mocks using `jest.resetAllMocks` in `beforeEach` instead of `beforeAll`, so that each `it` will have the utility mocks reset to the default `jest.fn()` return (undefined). Then in each `it` block you can set the `mockReturnValue` for the function that you need like `mocked(getSelectedText).mockReturnValue(text);`